chapter07/pxd/ex1.pl

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+use feature "say";
+
+say "Enter a couple of lines (Ctrl-D to exit):";
+
+while (<>) {
+    
+    # Will match 'freddie'
+    if (/fred/) { 
+    	say "The entered line contains the letters fred "; 
+    }
+    
+    # will match only 'fred' 
+    if ( /(^|\s)fred(\s|$)/ ) {
```

Comment:
Nice, I'm assuming that this is to show that you can do both (It does say to print the line if it mentions fred, so if you did it that way then with this you'd get duplicate lines).